generated/niscope/session.py

```diff
@@ -1822,10 +1822,10 @@ def fetch(self, num_samples=None, relative_to=enums.FetchRelativeTo.PRETRIGGER,
 
         for i in range(len(wfm_info)):
             start = i * num_samples
+            end = start + wfm_info[i].actual_samples
```

Comment:
Would we be able to get rid of the `actual_samples` field on `WaveformInfo` if you did this work before the transformation in `_fetch`?

Referring to:
```python
return waveform_array, [waveform_info.WaveformInfo(wfm_info_ctype[i]) for i in range(self._actual_num_wfms())]
```

Reply:
`actual_samples` is in the struct that we get back from the driver.

All the different `_fetch` functions are completely code generated. In order to do something there
we would have to torque around the code generator for a special case in scope, or have even more hand maintained functions where we do this same manipulation in each one. I don't think either of those options are very good.

Reply:
Understandable.

Since this is purely an implementation detail, I'm just generally concerned about publicly exposing the,

What if we `_` prefix `actual_samples`?

Reply:
Updated to make `actual_samples` "private"